src/core/mount.c

```diff
@@ -1402,8 +1402,6 @@ static int mount_setup_unit(
 
         u = manager_get_unit(m, e);
         if (!u) {
-                delete = true;
-
```

Comment:
oh, sorry, I was wrong. We can't remove that.
For example:
```c
                 r = unit_new_for_name(m, sizeof(Mount), e, &u);
                  if (r < 0)
                          goto fail;
```
We should `fail: if (delete && u)  unit_free(u)`(or we have a memory leak)
I think the fix is:
```c
diff --git a/src/core/device.c b/src/core/device.c
index 074e93f..8e2e3c7 100644
--- a/src/core/device.c
+++ b/src/core/device.c
@@ -359,7 +359,7 @@ static int device_setup_unit(Manager *m, struct udev_device *dev, const char *pa
 fail:
         log_unit_warning_errno(u, r, "Failed to set up device unit: %m");

-        if (delete)
+        if (delete && u)
                 unit_free(u);

         return r;
```
This will prevent that assert if we add some code to the `device_setup_unit`